### PR TITLE
53 match i

### PR DESCRIPTION
**Pull request by jalew188**
Add docs for `alpharaw.match.match_utils` and `alpharaw.match.spec_finder`.

### CODE REVIEW

File: alpharaw/match/spec_finder.py [hunk 1]
```diff
@@ -9,8 +11,31 @@ def find_spec_idxes_by_rt(
     query_stop_rt: float,
     query_left_mz: float,
     query_right_mz: float,
-):
+) -> np.ndarray:
+    """
+    Find MS2 spectrum indices (int32) from the `spectrum_df`
```
Comment (mschwoer) [inline alpharaw/match/spec_finder.py:16]: maybe comment (here or somewhere else) why int32 is set explicitly here

File: alpharaw/match/spec_finder.py [hunk 2]
```diff
@@ -84,7 +156,34 @@ def find_spec_idxes(
     query_stop_rt: float,
     query_left_mz: float,
     query_right_mz: float,
-):
+) -> np.ndarray:
+    """
+    Find MS2 spectrum indices (int32) from the all spectra
```
Comment (mschwoer) [inline alpharaw/match/spec_finder.py:161]: `from the spectra`? (also below)

**Review by mschwoer — APPROVED** [review]

File: alpharaw/match/spec_finder.py (hunk 2 above)
Comment (jalew188) [reply alpharaw/match/spec_finder.py:161]: -:)

File: alpharaw/match/spec_finder.py (hunk 1 above)
Comment (jalew188) [reply alpharaw/match/spec_finder.py:16]: Done

**Pull request merged**